.tasks/lint.yml

```
@@ -0,0 +1,37 @@
+version: '3'
+
+vars:
+  CLI_ARGS: '{{.CLI_ARGS | default ""}}'
+  GOFUMPT_VERSION: "v0.7.0"
+  GOLANGCI_LINT_VERSION: "v2.2.1"
+
+tasks:
+  install:
+    desc: "Install linting tools"
+    cmds:
+      - echo "Installing gofumpt..."
+      - go install mvdan.cc/gofumpt@{{.GOFUMPT_VERSION}}
+      - echo "Installing golangci-lint..."
+      - go install github.com/golangci/golangci-lint/v2/cmd/golangci-lint@{{.GOLANGCI_LINT_VERSION}}
+      - echo "Linting tools installed successfully"
+
+  default:
+    desc: "Run all code quality checks"
+    cmds:
+      - echo "Running linting..."
+      - |
+        if [ "$BUILDKITE" = "true" ]; then
+          echo "Running in BuildKite environment with buildvcs=false"
+          GOFLAGS="-buildvcs=false" golangci-lint run --timeout=5m {{.CLI_ARGS}}
+        else
+          echo "Running in local environment"
+          golangci-lint run {{.CLI_ARGS}}
+        fi
+      - echo "Linting completed successfully"
+
+  fix:
+    desc: "Run linting with auto-fix"
+    cmds:
+      - task: lint
```

There is problem with calling task named `lint`, as it does not exist.
```
task lint:fix
task: Failed to run task "lint:fix": task: Task "lint:lint" does not exist
```

```suggestion
      - task: default
```